### ISSUE TITLE
Support for cascading replicas

### ISSUE THREAD

**Issue by twingate-blee** [terraform-google-modules/terraform-google-sql-db#502]
### TL;DR

Cloud SQL now supports cascading replicas (https://cloud.google.com/sql/docs/postgres/replication/#cascading-replicas) for postgres and mysql. With the current module we can only create a replica at one level. 

### Terraform Resources

google_sql_database_instance

### Detailed design

We should be able to add up to four levels of replicas, including the primary instance.

My thought is that the existing variable `read_replicas` can be nested to contain multiple levels of replicas

In the resource
```
resource "google_sql_database_instance" "replicas" {

```

The variable `master_instance_name` will need to be dynamically set instead of always referencing the primary instance `google_sql_database_instance.default.name`

**Comment by bertekintw** [comment]
It will be great if this support is added.

**Comment by g-awmalik** [comment]
@twingate-blee - if you're able to do a PR for this feature, I'll be happy to review it for you.

**Comment by imrannayer** [comment]
@twingate-blee @bertekintw In version 20 of this module we added a feature which allows you to create replica in a different way. You can follow [this example](https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/examples/postgresql-with-cross-region-failover) and create cascading replica.
